### PR TITLE
[ops, model] feat: forward temperature and return entropy from log-probs path

### PR DESCRIPTION
**Pull request by Luosuu**
## Summary

Extends PR #711's per-token log-probs path so VeOmni-built models can fully replace verl's `FSDPEngineWithLMHead.patch_forward_with_backends` monkey patch:

- Forwards `temperature` from `ForCausalLMLoss` into `chunk_logprobs_function` (verl's PPO actor divides logits by `T` before log-softmax).
- Returns per-token softmax entropy alongside `log_probs` so `output.entropy` populates verl's `prepare_model_outputs(use_fused_kernels=True, calculate_entropy=True)` consumer (`entropy_rmpad = output.entropy.squeeze(0)`).
- Bumps the wrapper return shape from `(loss, logits, log_probs)` → `(loss, logits, log_probs, entropy)`. Every in-tree modeling + patchgen config moved to the uniform unpack `loss, _, log_probs, entropy = self.loss_function(...)` and assigns `output.entropy = entropy` next to the existing `output.log_probs` line.

## Implementation notes

- **Kernel** (`veomni/ops/kernels/cross_entropy/chunk_logprobs.py`): forward computes entropy `H[p] = logsumexp(logits) − Σ p · logits` with verl's op order; backward gates the entropy-grad path on `dentropy is not None` so the loss-only path pays no extra FLOPs. IGN-position masking applies to entropy too.
- **Wrapper** (`veomni/ops/kernels/cross_entropy/__init__.py`): pops `temperature` (default 1.0) and `return_log_probs` from kwargs; `_chunk_loss_dispatch` shim mirrors the same 4-tuple contract. `ForSequenceClassificationLoss` returns `(loss, logits, None, None)` for unpack uniformity.
- **Output dataclass** (`veomni/utils/model_outputs.py`): `CausalLMOutputWithLogProbs` gains `entropy: Optional[Tensor]`.
- **Modelings**: 12 modeling files + 11 patchgen configs updated; all `generated/*` regenerated via `python -m veomni.patchgen.check_patchgen --fix`.

## Test plan

- [x] `pytest tests/ops/test_chunk_logprobs.py -v` — 10/10 pass; new tests cover `temperature ∈ {1.0, 0.7, 2.5}` bitwise-parity vs verl's `FusedLinearForPPOFunction` on log_probs, entropy, and gradients
- [x] `pytest tests/ops/test_seqcls_loss.py -v` — 6/6 pass
- [x] `pytest tests/models/test_return_log_probs_e2e.py -v` — pins entropy shape, IGN masking, non-negative valid positions, and the verl consumer contract (`output.entropy.shape == labels.shape`)
- [x] `make quality` clean
- [x] `python -m veomni.patchgen.check_patchgen` clean (all 30+ generated files regenerated and verified)

### CODE REVIEW

File: veomni/ops/kernels/cross_entropy/chunk_logprobs.py [hunk 1]
```diff
@@ -196,14 +224,29 @@ def backward(ctx, dlog_probs: Optional[torch.Tensor]):
...
+            # ── entropy gradient path (mirrors verl exactly) ─────────────
+            # ∂H[p]/∂logits[i, j] = p_j (log p_j + H[p]) so
+            # dlogits[i, j] = -dentropy[i] * p_j * (log p_j + H[p_i])
+            # The ``-`` sign is folded into the ``-dentropy.unsqueeze(-1)``
+            # multiplier (matches verl's expression).
+            if dent_chunk is not None:
+                log_probs_full = logits.log_softmax(dim=-1)
+                entropy_full = torch.logsumexp(logits, dim=-1) - torch.sum(probs * logits, dim=-1)
+                masked_dent = (dent_chunk * mask).unsqueeze(-1)
+                dlogits = dlogits + probs * (log_probs_full + entropy_full.unsqueeze(-1)) * (-masked_dent)
```
[the remark this answers is not in the document]
Comment (Luosuu) [reply veomni/ops/kernels/cross_entropy/chunk_logprobs.py:249]: Thanks for flagging this. Keeping the current form intentionally — bitwise parity with verl's `_fused_linear_for_ppo_bwd` (`verl/utils/experimental/torch_functional.py:73-77`) is what `test_bitwise_parity_with_verl_fused_linear_for_ppo` (parametrized over T ∈ {1.0, 0.7, 2.5}) is pinning, and the suggested rewrite — while algebraically equivalent — is not fp32-bitwise-equal because the `logsumexp + (-logsumexp)` cancellation in our path doesn't round identically to a direct `z - sum(p*z)`.

The `0 * -inf = NaN` concern doesn't materialize in practice. PyTorch's `log_softmax` is the stable `z - logsumexp(z)` form, so it returns finite values (e.g. `-1e8`) where `softmax` has underflowed to `0`, not `-inf`. Quick check with pathological logits `[0., 0., -200., 1e8]` (three entries underflow to softmax=0):

```
log_softmax: tensor([-1.0000e+08, -1.0000e+08, -1.0000e+08,  0.0000e+00])
probs * (log_probs + entropy): tensor([-0., -0., -0., 0.])  # no NaN
```

The only way the product hits NaN is if input logits already contain `-inf` (broken upstream). Given (a) bitwise parity with verl is a non-negotiable test pin and (b) the NaN window is theoretical for well-formed fp32 logits, the marginal stability win doesn't justify the parity loss.

File: tests/ops/test_chunk_logprobs.py [hunk 2]
```diff
@@ -157,61 +170,108 @@ def test_bitwise_parity_with_reference():
...
+        one_hot = torch.zeros_like(probs).scatter_(-1, safe, 1.0)
+        masked_dlp = (dlp_flat * mask).unsqueeze(-1)
+        dlogits = dlogits + masked_dlp * (one_hot - probs)
+
+    if grad_entropy is not None:
+        dent_flat = grad_entropy[..., :-1].reshape(-1)
+        log_probs_full = logits.log_softmax(dim=-1)
+        entropy_full = torch.logsumexp(logits, dim=-1) - torch.sum(probs * logits, dim=-1)
+        masked_dent = (dent_flat * mask).unsqueeze(-1)
+        dlogits = dlogits + probs * (log_probs_full + entropy_full.unsqueeze(-1)) * (-masked_dent)
```
[the remark this answers is not in the document]
Comment (Luosuu) [reply tests/ops/test_chunk_logprobs.py:270]: Same rationale as the kernel-side comment — the manual reference must mirror the kernel's exact op order (`probs * (log_probs + entropy) * (-masked_dent)`) so the test pins the kernel against verl bitwise. If the test reference drifts to `probs * (sum_p_z - z) * masked_dent`, it stops being a verl-parity reference and starts being a different algebraically-equivalent reference, which would (a) loosen the contract and (b) make the verl-parity test redundant with the reference test.

The NaN concern likewise doesn't fire here for well-formed fp32 logits (PyTorch's `log_softmax` is the stable `z - logsumexp` form).

**Review by TimYangst — APPROVED** [review]

File: veomni/models/transformers/qwen3_vl/qwen3_vl_gpu_patch_gen_config.py [hunk 3]
```diff
@@ -1108,10 +1108,11 @@ def qwen3_vl_for_conditional_generation_forward_patched(
     loss = None
     logits = None
     log_probs = None
+    entropy = None
     if labels is not None:
         # Modification: OpSlot guard for cross-entropy loss.
         if veomni_causal_lm_loss.use_non_eager_impl:
-            loss, logits, log_probs = veomni_causal_lm_loss(
+            loss, logits, log_probs, entropy = veomni_causal_lm_loss(
```
Comment (Coach257) [inline veomni/models/transformers/qwen3_vl/qwen3_vl_gpu_patch_gen_config.py:1115]: better to import veomni_causal_lm_loss at the head. Though not cause runtime error, IDE will label this as undefined.
